stdlib/public/CMakeLists.txt

```diff
@@ -191,6 +191,12 @@ if(SWIFT_SHOULD_BUILD_EMBEDDED_STDLIB)
     )
   endif()
 
+  if("WebAssembly" IN_LIST LLVM_TARGETS_TO_BUILD)
+    list(APPEND EMBEDDED_STDLIB_TARGET_TRIPLES
+      "wasm32    wasm32-none-none-wasm    wasm32-none-none-wasm"
```

Comment:
`none` is not a valid vendor, it should be `wasm32-unknown-none-wasm`.

Reply:
I'm following a precedent established here for other triples, which all use `none` as vendor.

Reply:
To clarify, I'm ok with using `unknown` for the vendor component, but IMO that deserves a separate PR from whoever is relying on existing `none` triples. I don't think I can change them all to use `unknown` without breakage at this point within this current PR.

Reply:
@kubamracek correct me if I am wrong: but I was under the impression that the `none` in the triples was for an indication of the vendor of the OS, not the vendor of the cpu. Other parts of the target infrastructure through llvm end up taking different paths (the baremetal/freestanding route) when the vendor and os are none.

If however the vendor is supposed to be applied to the CPU (which wasm32 is a strange beast for that, being it is a virtualized cpu) then we should, as Max says, address that in its own commits.

Reply:
@MaxDesiatov can you clarify what "other triples" is referring to?  There are definitely existing triples that use the unknown vendor.  This is a *very* established precedent (of using `unknown` for the vendor), far more than the `none`.

Reply:
I'm referring to triples used in this same file just above the lines being added: https://github.com/apple/swift/pull/70629/files#diff-5d38339db2a49b611cf5a57fed0a38e0a24ce535a97400cc730bd9ea1f96753bR156-R192

Reply:
🤦

Reply:
https://github.com/apple/swift/pull/70682